src/conf_mode/protocols_rip.py

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2020 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from sys import exit
+
+from vyos import ConfigError
+from vyos.config import Config
+from vyos.util import call
+from vyos.template import render
+
+from vyos import airbag
+airbag.enable()
+
+config_file = r'/tmp/ripd.frr'
+
+def get_config():
+    conf = Config()
+    rip_conf = {
+        'rip_conf'          : False,
+        'default_distance'  : None,
+        'default_originate' : False,
+        'old_rip'  : {
+            'default_metric'  : None,
+            'distribute'      : {},
+            'neighbors'       : {},
+            'networks'        : {},
+            'net_distance'    : {},
+            'passive_iface'   : {},
+            'redist'          : {},
+            'route'           : {},
+            'ifaces'          : {},
+            'timer_garbage'   : 120,
+            'timer_timeout'   : 180,
+            'timer_update'    : 30
+        },
+        'rip'  : {
+            'default_metric'   : None,
+            'distribute'       : {},
+            'neighbors'        : {},
+            'networks'         : {},
+            'net_distance'     : {},
+            'passive_iface'    : {},
+            'redist'           : {},
+            'route'            : {},
+            'ifaces'           : {},
+            'timer_garbage'    : 120,
+            'timer_timeout'    : 180,
+            'timer_update'     : 30
+        }
+    }
+
+    if not (conf.exists('protocols rip') or conf.exists_effective('protocols rip')):
```

Comment:
It would be better to define `base = ['protocols', 'rip']` and the reference that variable when you need to change the config hirarchy, this also makes refactoring easier, as only one place need to be altered.

Reply:
done
Add 
```
     base = ['protocols', 'rip'] 
```